### ISSUE TITLE
print-lldp.c: Incorrect printing of 'Protocol ID' field for organizational TLVs

### ISSUE THREAD

**Issue by skini-bsn** [the-tcpdump-group/tcpdump#511]
For Organizational specific TLV (127) in an LLDP packet with the 'Application Priority Table' subtype 12, the 'Protocol ID' field is incorrectly parsed/printed. 
Sample output-

```
00:54:28.844088 LLDP, length 161
        Chassis ID TLV (1), length 7
          Subtype MAC address (4): 00:00:00:02:00:02 (oui Ethernet)
          0x0000:  0400 0000 0200 02
        Port ID TLV (2), length 13
          Subtype Interface Name (5): leaf0b-eth10
          0x0000:  056c 6561 6630 622d 6574 6831 30
        Time to Live TLV (3), length 2: TTL 120s
          0x0000:  0078
        Port Description TLV (4), length 41: Big Cloud Fabric Switch Port leaf0b-eth10
          0x0000:  4269 6720 436c 6f75 6420 4661 6272 6963
          0x0010:  2053 7769 7463 6820 506f 7274 206c 6561
          0x0020:  6630 622d 6574 6831 30
        System Name TLV (5), length 6: leaf0b
          0x0000:  6c65 6166 3062
        System Description TLV (6), length 17
          5c:16:c7:00:00:01
          0x0000:  3563 3a31 363a 6337 3a30 303a 3030 3a30
          0x0010:  31
        Organization specific TLV (127), length 5: OUI Unknown (0x0026e1)
          0x0000:  0026 e101 01
        Organization specific TLV (127), length 9: OUI Unknown (0x0026e1)
          0x0000:  0026 e102 6c65 6166 30
        Organization specific TLV (127), length 5: OUI Unknown (0x0026e1)
          0x0000:  0026 e103 01
        Organization specific TLV (127), length 16: OUI Unknown (0x0026e1)
          0x0000:  0026 e104 0000 5c16 c70b ba1b 0000 0000
        Organization specific TLV (127), length 6: OUI Ethernet bridged (0x0080c2)
          Priority Flow Control Configuration Subtype (11)
            Willing: 0, MBC: 0, RES: 0, PFC cap:1 
            PFC Enable
             Priority : 0  1  2  3  4  5  6  7
             Value    : 0  0  0  0  1  0  0  0  
          0x0000:  0080 c20b 0110
        Organization specific TLV (127), length 8: OUI Ethernet bridged (0x0080c2)
          Application Priority Subtype (12)
            RES: 0
            Application Priority Table
              Priority: 4, RES: 0, Sel: 4Protocol ID: 33804
          0x0000:  0080 c20c 0084 0cbc
        End TLV (0), length 0
        0x0000:  0207 0400 0000 0200 0204 0d05 6c65 6166
        0x0010:  3062 2d65 7468 3130 0602 0078 0829 4269
        0x0020:  6720 436c 6f75 6420 4661 6272 6963 2053
        0x0030:  7769 7463 6820 506f 7274 206c 6561 6630
        0x0040:  622d 6574 6831 300a 066c 6561 6630 620c
        0x0050:  1135 633a 3136 3a63 373a 3030 3a30 303a
        0x0060:  3031 fe05 0026 e101 01fe 0900 26e1 026c
        0x0070:  6561 6630 fe05 0026 e103 01fe 1000 26e1
        0x0080:  0400 005c 16c7 0bba 1b00 0000 00fe 0600
        0x0090:  80c2 0b01 10fe 0800 80c2 0c00 840c bc00
        0x00a0:  00
```

In the above packet, the last organization TLV the "Application Priority Table" value is '84 0c bc' (last but one line). 0x84 is correctly parsed in the output to represent 'Priority: 4, RES: 0, Sel: 4'. However, the protocol ID field value is '0x0c 0xbc' (last two bytes). But instead tcpdump does not skip the first byte and prints '0x84 0x0c' = 33804 in the output.

Suggested patch-

```
diff --git a/print-lldp.c b/print-lldp.c
index 82b7ebe..d68e6d5 100644
--- a/print-lldp.c
+++ b/print-lldp.c
@@ -778,9 +778,9 @@ lldp_private_8021_print(netdissect_options *ndo,
         ND_PRINT((ndo, "\n\t    Application Priority Table"));
         while(i<sublen) {
                tval=*(tptr+i+5);
-               ND_PRINT((ndo, "\n\t      Priority: %d, RES: %d, Sel: %d",
+               ND_PRINT((ndo, "\n\t      Priority: %d, RES: %d, Sel: %d ",
                         tval >> 5, (tval >> 3) & 0x03, (tval & 0x07)));
-               ND_PRINT((ndo, "Protocol ID: %d", EXTRACT_16BITS(tptr + i + 5)));
+               ND_PRINT((ndo, "Protocol ID: %d", EXTRACT_16BITS(tptr + i + 6)));
                i=i+3;
         }
         break;
```

**Comment by fxlb** [comment]
@skini-bsn
Thanks for the report. Could you give a pcap file for test containing  a packet with TVL 127/subtype 12 in ascii hexdump form (converted with `xxd` for example) ?

**Comment by fxlb** [comment]
Fixed by @fenner with dc36fbcfb329504ace442ab0b79dd85819dfea17.
